packages/loopring_v3/contracts/core/impl/libexchange/ExchangeBlocks.sol

```diff
@@ -201,72 +201,93 @@ library ExchangeBlocks
                 timestamp: header.timestamp
             });
 
+            ExchangeData.AuxiliaryData[] memory block_auxiliaryData;
```

Comment:
Brecht, it seems we still need a `ExchangeData.AuxiliaryData[]` in memory, aren't we still decoding the aux data for all conditional transactions? What do you mean “unnecessary decoding done by solidity can be avoided.”？

Reply:
There are additional hidden costs for complex data structures that solidity does behind the scees, and I wasn't aware it could be that expensive. The main cost that is avoided by passing the data as `bytes` is that solidity does not need to do all the abi decoding -> memory for the complete array. `abi.decode` is quite expensive especially when working on data in memory because a copy is made of all the data and of course the parsing of the abi encoded data. By just telling solidity to handle it like a simple `bytes` array there is very little decoding to do so it's much cheaper.